common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -138,15 +165,25 @@ protected void serviceInit(Configuration conf) throws Exception {
       registeredExecutorFile =
         new File(getRecoveryPath().toUri().getPath(), RECOVERY_FILE_NAME);
 
+      secretsFile = new File(getRecoveryPath().toUri().getPath(), SECRETS_RECOVERY_FILE_NAME);
```

Comment:
e.g. you can rename `reloadCreds` to `createSecretManager` or something, and then it would contain all the code to re-load existing state from the db, and would only be triggered if needed (i.e. in L149).